From 69e956b612ea3bf7513db4f6cbdc8a3c9c2e93e7 Mon Sep 17 00:00:00 2001
From: nushanth <nushanth@qti.qualcomm.com>
Date: Thu, 12 Mar 2026 19:40:10 +0530
Subject: [PATCH] Fix for Backhaul manager FSM and Early AP Report when BSTA
 Reconnect within the WIRELESS_WAIT_FOR_RECONNECT timeout

 Move Backhaul manager FSM to CONNECTED and Reset the
 early_ap_capability_report_sent flag to false when BSTA
 reconnect within the WIRELESS_WAIT_FOR_RECONNECT timeout

Signed-off-by: nushanth <nushanth@qti.qualcomm.com>
---
 .../beerocks/slave/backhaul_manager/backhaul_manager.cpp  | 8 +++-----
 .../beerocks/slave/tasks/ap_autoconfiguration_task.cpp    | 3 +++
 2 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/agent/src/beerocks/slave/backhaul_manager/backhaul_manager.cpp b/agent/src/beerocks/slave/backhaul_manager/backhaul_manager.cpp
index eb03ec7..9638bf7 100644
--- a/agent/src/beerocks/slave/backhaul_manager/backhaul_manager.cpp
+++ b/agent/src/beerocks/slave/backhaul_manager/backhaul_manager.cpp
@@ -2447,14 +2447,12 @@ bool BackhaulManager::hal_event_handler(bwl::base_wlan_hal::hal_event_ptr_t even
         } else if (FSM_IS_IN_STATE(WIRELESS_WAIT_FOR_RECONNECT)) {
             LOG(DEBUG) << "reconnected successfully, continuing";
 
-            // IRE running controller
-            if (db->device_conf.local_controller && !db->device_conf.local_gw) {
-                FSM_MOVE_STATE(CONNECTED);
-            } else {
+            if (!(db->device_conf.local_controller && !db->device_conf.local_gw)) {
                 // send topology_notification after reconnect
                 m_task_pool.send_event(eTaskType::TOPOLOGY, TopologyTask::eEvent::AGENT_RADIO_STATE_CHANGED);
-                FSM_MOVE_STATE(OPERATIONAL);
             }
+
+            FSM_MOVE_STATE(CONNECTED);
         }
     } break;
 
diff --git a/agent/src/beerocks/slave/tasks/ap_autoconfiguration_task.cpp b/agent/src/beerocks/slave/tasks/ap_autoconfiguration_task.cpp
index c01c980..4979827 100644
--- a/agent/src/beerocks/slave/tasks/ap_autoconfiguration_task.cpp
+++ b/agent/src/beerocks/slave/tasks/ap_autoconfiguration_task.cpp
@@ -325,6 +325,9 @@ void ApAutoConfigurationTask::handle_event(uint8_t event_enum_value, const void
             m_discovery_status[radio->wifi_channel.get_freq_type()].completed = false;
             m_discovery_status[radio->wifi_channel.get_freq_type()].msg_sent = false;
 
+            //change early capability report sent to false so that it is sent again
+            db->controller_info.early_ap_capability_report_sent = false;
+
             LOG(DEBUG) << "starting discovery sequence on radio_iface=" << radio->front.iface_name;
             FSM_MOVE_STATE(radio->front.iface_name, eState::CONTROLLER_DISCOVERY);
             m_task_is_active = true;
-- 
2.34.1

